### PR TITLE
update YOLO series TRT eval demo

### PR DESCRIPTION
**Pull request by yghstill**

### CODE REVIEW

File: example/auto_compression/pytorch_yolo_series/README.md [hunk 1]
```diff
@@ -136,13 +136,27 @@ python eval.py --config_path=./configs/yolov7_tiny_qat_dis.yaml
...
 
-执行完自动压缩后会默认在`save_dir`中生成`quant_model.onnx`的ONNX模型文件，可以直接使用TensorRT测试脚本进行验证。
+加载`quant_model.onnx`和`calibration.cache`，可以直接使用TensorRT测试脚本进行验证，详细代码可参考[./TensorRT]
 
 - 进行测试：
 ```shell
-python yolov7_onnx_trt.py --model_path=output/quant_model.onnx --image_file=images/000000570688.jpg --precision=int8
+cd TensorRT
+python trt_eval.py --onnx_model_file=yolov7_tiny_quant_onnx/yolov7_tiny_quant.onnx \
+                   --calibration_file=yolov7_tiny_quant_onnx/calibration.cache \
```
Comment (wanghaoshuang) [inline example/auto_compression/pytorch_yolo_series/README.md]: onnx_model_file、 calibration_file和第4节给出的产出文件路径不一致。

File: example/auto_compression/pytorch_yolo_series/TensorRT/README.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+```shell
+python trt_eval.py --onnx_model_file=yolov7_tiny_quant_onnx/yolov7_tiny_quant.onnx \
+                   --calibration_file=yolov7_tiny_quant_onnx/calibration.cache \
+                   --image_file=../images/000000570688.jpg \
+                   --precision_mode=int8
+```
+
+### FAQ
+
+- 测试内存和显存占用时，首次运行会将ONNX模型转换成TRT模型，耗时不准确，再此运行trt.eval.py可获取真实的内存和显存占用。
```
Comment (wanghaoshuang) [inline example/auto_compression/pytorch_yolo_series/TensorRT/README.md]: trt_eval.py

File: example/auto_compression/pytorch_yolo_series/TensorRT/trt_backend.py [hunk 3]
```diff
@@ -0,0 +1,265 @@
...
+            print("[TRT Backend] Loading ONNX model ...")
+            onnx_model = onnx_model_file
+            if not isinstance(onnx_model_file, onnx.ModelProto):
+                onnx_model = onnx.load(onnx_model_file)
+            onnx_model = remove_initializer_from_input(onnx_model)
+            if not parser.parse(onnx_model.SerializeToString()):
+                print("ERROR: Failed to parse the ONNX file.")
+                for error in range(parser.num_errors):
+                    print(parser.get_error(error))
+                sys.exit(0)
```
Comment (wanghaoshuang) [inline example/auto_compression/pytorch_yolo_series/TensorRT/trt_backend.py]: 报错退出，原则上不能返回0吧，throw exception是不是更合理些？

File: example/auto_compression/pytorch_yolo_series/TensorRT/trt_backend.py [hunk 4]
```diff
@@ -0,0 +1,265 @@
...
+
+            if shape_info is None:
+                builder.max_batch_size = 1
+                for i in range(len(onnx_model.graph.input)):
+                    input_shape = [
+                        x.dim_value
+                        for x in onnx_model.graph.input[0]
```
Comment (wanghaoshuang) [inline example/auto_compression/pytorch_yolo_series/TensorRT/trt_backend.py:151]: input[I]? 当前i貌似没被用到。

File: example/auto_compression/pytorch_yolo_series/TensorRT/trt_backend.py [hunk 5]
```diff
@@ -0,0 +1,265 @@
...
+            else:
+                max_batch_size = 1
+                if shape_info is not None:
+                    assert len(
+                        shape_info
+                    ) == network.num_inputs, "Length of shape_info: {} is not same with length of model input: {}".format(
+                        len(shape_info), network.num_inputs)
+                    profile = builder.create_optimization_profile()
+                    for k, v in shape_info.items():
+                        if v[2][0] > max_batch_size:
```
Comment (wanghaoshuang) [inline example/auto_compression/pytorch_yolo_series/TensorRT/trt_backend.py:166]: v[2][0]是什么意思？在当前方法添加对shape_info的解释，或者用变量表示一下这些魔法数字。

File: example/auto_compression/pytorch_yolo_series/TensorRT/trt_backend.py [hunk 6]
```diff
@@ -0,0 +1,265 @@
...
+
+        self.context = self.engine.create_execution_context()
+        if shape_info is not None:
+            self.context.active_optimization_profile = 0
+        self.stream = cuda.Stream()
+        self.bindings = []
+        self.inputs = []
+        self.outputs = []
+        for binding in self.engine:
+            self.bindings.append(0)
```
Comment (wanghaoshuang) [inline example/auto_compression/pytorch_yolo_series/TensorRT/trt_backend.py:196]: 这一句的作用是？self.bindings只是一个元素为0的数组？是指默认的device?

File: example/auto_compression/pytorch_yolo_series/TensorRT/trt_eval.py [hunk 7]
```diff
@@ -0,0 +1,321 @@
...
+            img,
+            text, (x0, y0 + txt_size[1]),
+            font,
+            0.8, (0, 255, 0),
+            thickness=2)
+
+    return img
+
+
+def get_current_memory_mb():
```
Comment (wanghaoshuang) [inline example/auto_compression/pytorch_yolo_series/TensorRT/trt_eval.py:167]: 可以迁移到common package么？

File: example/auto_compression/pytorch_yolo_series/README.md (hunk 1 above)
Comment (yghstill) [reply example/auto_compression/pytorch_yolo_series/README.md]: done

File: example/auto_compression/pytorch_yolo_series/TensorRT/README.md (hunk 2 above)
Comment (yghstill) [reply example/auto_compression/pytorch_yolo_series/TensorRT/README.md]: done

File: example/auto_compression/pytorch_yolo_series/TensorRT/trt_backend.py (hunk 3 above)
Comment (yghstill) [reply example/auto_compression/pytorch_yolo_series/TensorRT/trt_backend.py]: done

File: example/auto_compression/pytorch_yolo_series/TensorRT/trt_backend.py (hunk 4 above)
Comment (yghstill) [reply example/auto_compression/pytorch_yolo_series/TensorRT/trt_backend.py:151]: input[i]在下面用到了

File: example/auto_compression/pytorch_yolo_series/TensorRT/trt_eval.py (hunk 7 above)
Comment (yghstill) [reply example/auto_compression/pytorch_yolo_series/TensorRT/trt_eval.py:167]: next pr

File: example/auto_compression/pytorch_yolo_series/TensorRT/trt_backend.py (hunk 6 above)
Comment (yghstill) [reply example/auto_compression/pytorch_yolo_series/TensorRT/trt_backend.py:196]: next pr

File: example/auto_compression/pytorch_yolo_series/TensorRT/trt_backend.py (hunk 5 above)
Comment (yghstill) [reply example/auto_compression/pytorch_yolo_series/TensorRT/trt_backend.py:166]: next pr

**Review by wanghaoshuang — APPROVED** [review]

**Comment by xiyangyang99** [comment]
我用我自己的本地数据集验证和coco数据集验证的时候，报错了！是同样的错误。
loading annotations into memory...
Done (t=0.30s)
creating index...
index created!
[TRT Backend] Use FP16.
[TRT Backend] Loading ONNX model ...
[05/11/2023-17:28:57] [TRT] [W] onnx2trt_utils.cpp:366: Your ONNX model has been generated with INT64 weights, while TensorRT does not natively support INT64. Attempting to cast down to INT32.
[TRT Backend] Completed parsing of ONNX file.
[TRT Backend] Building an engine from onnx model may take a while...
[05/11/2023-17:28:58] [TRT] [W] TensorRT was linked against cuBLAS/cuBLAS LT 11.6.3 but loaded cuBLAS/cuBLAS LT 11.5.2
[05/11/2023-17:39:57] [TRT] [W] TensorRT was linked against cuBLAS/cuBLAS LT 11.6.3 but loaded cuBLAS/cuBLAS LT 11.5.2
[TRT Backend] Start Creating Engine.
[05/11/2023-17:39:57] [TRT] [W] TensorRT was linked against cuBLAS/cuBLAS LT 11.6.3 but loaded cuBLAS/cuBLAS LT 11.5.2
[TRT Backend] Completed Creating Engine.
[05/11/2023-17:39:57] [TRT] [W] TensorRT was linked against cuBLAS/cuBLAS LT 11.6.3 but loaded cuBLAS/cuBLAS LT 11.5.2
[TRT Backend] Completed TrtEngine init ...
Evaluation stage, Run batch:|                                            | 0/619W0511 17:39:57.326681 1460977 gpu_resources.cc:61] Please NOTE: device: 0, GPU Compute Capability: 8.6, Driver API Version: 11.4, Runtime API Version: 11.4
W0511 17:39:57.326769 1460977 gpu_resources.cc:91] device: 0, cuDNN Version: 8.7.
Evaluation stage, Run batch:|                                            | 0/619
Traceback (most recent call last):
  File "/home/bowen/下载/1-project/TensorRT-For-YOLO-Series/PaddleSlim/example/auto_compression/pytorch_yolo_series/TensorRT/trt_eval.py", line 309, in <module>
    main()
  File "/home/bowen/下载/1-project/TensorRT-For-YOLO-Series/PaddleSlim/example/auto_compression/pytorch_yolo_series/TensorRT/trt_eval.py", line 299, in main
    eval()
  File "/home/bowen/下载/1-project/TensorRT-For-YOLO-Series/PaddleSlim/example/auto_compression/pytorch_yolo_series/TensorRT/trt_eval.py", line 227, in eval
    for data in val_loader:
  File "/home/bowen/下载/yes/envs/yolo/lib/python3.8/site-packages/paddle/fluid/dataloader/dataloader_iter.py", line 276, in __next__
    data = self._reader.read_next_list()
OSError: (External) CUDA error(400), invalid resource handle. 
  [Hint: Please search for the error code(400) on website (https://docs.nvidia.com/cuda/cuda-runtime-api/group__CUDART__TYPES.html#group__CUDART__TYPES_1g3f51e3575c2178246db0a94a430e0038) to get Nvidia's official solution and advice about CUDA Error.] (at /paddle/paddle/fluid/operators/reader/buffered_reader.cc:235)
Process finished with exit code 1


请问是cuda版本的问题吗？cudaerror 400 的问题怎么解决呀？

**Pull request merged**